vignettes/pk-analysis.Rmd

```diff
@@ -90,3 +94,85 @@ exportPKAnalysesToCSV(pkAnalyses = pkAnalysis, filePath = csvPKAnalysisPath)
+It is not possible at the moment to define your own `formula`.
+
+The utility `addUserDefinedPKParameter` creates and returns an instance of the `UserDefinedPKParameter` which can then be parameterized
+for specific requirements.
+
+### Examples
+
+#### Calculate AUC between for a specific time interval.
+
+```{r UserDefinedTimeInterval}
+# Adds a user defined parameter named `MyAuc` that will calculate the value of AUC between t=50 min and t=80min only.
+
+# Create a new parameter based on the standard AUC parameter
+myAUC <- addUserDefinedPKParameter(
+   name = "MyAUC",
+   standardPKParameter = StandardPKParameter$AUC_tEnd
+)
+
+# Specifies start time and end time in minute
+myAUC$startTime <- 50
+myAUC$endTime <- 80
+```
+
+#### Calculate CMax between the 4th and 5th application of a multiple applications simulation.
+
+```{r UserDefinedApplicationInterval}
+# Adds a user defined parameter named `MyCMax` that will calculate the value of Cmax between the 4th and 5th application
+
+# Create a new parameter based on the standard C_max parameter
+ myCMax <- addUserDefinedPKParameter(
+   name = "MyCMax",
+   standardPKParameter = StandardPKParameter$C_max
+)
+
+# Specifies start application (4th) and end applicaiton (5th)
+ myCMax$startApplicationIndex <- 4
+ myCMax$endApplicationIndex <- 5
+```
+
+#### Update Dimension of a standard parameter for correct visualization
+
+PK parameters currently always assume that the underlying quantity is a concentration and do not check for units.
+If the quantity of interest has another dimension, creating a user-defined PK paramter `AND` set its display unit to any valid unit of the target dimension can do the trick 
+
+
+```{r UserDefinedDimensionChange}
+# Let's assume there is an observer called Q_observer in mg/m2 using the dimension Dose per body surface area. 
+# Simply using C_max would result in the parameter being shown in umol\l. 
+# To see the correct unit and dimension, the following can be done:
+
+QMax = addUserDefinedPKParameter(
+  name = "Q_max", 
+  standardPKParameter = StandardPKParameter$C_max, 
+  displayName = "Q max", 
+  displayUnit = "mg/m²"
+)
+```
+
+What will happen here:
+
+* The dimension of the user defined PK PArameter will be estimated from the display unit, e.g in this case `mg/m²`
+* The calculated value of the user defined PK Parameter will be interpreted as value in the base unit of this dimension, e.g. here "kg/dm²"
+* If the display unit differs from the base unit of the dimension, calculated value will be converted into the target display unit
+
+E.g. in the example above:
+
+* First, the C_max value of the quantity will be calculated (because the user defined parameter is derived from C_max)
+* From the display unit `mg/m²`, the dimension will be estimated ("Dose per body surface area")
+* The calculated C_max value will be interpreted as a value in `kg/dm²`
+* Finally, the calculated value will be converted to `mg/m²` and set in the user defined PK Parameter.
+
```

Comment:
Very good explanation and example!
This is one of the core features with userDef pk params which is not self-explaining at all

Reply:
that's your explanation :)

Reply:
really? cool :)